### PR TITLE
feat: support "umd" format

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

TODO:

`umdModuleName`  lacks some advanced features:

1. non string type
2. auto transform to camel case

We might implement them after understand why they're supported in https://modernjs.dev/module-tools/api/config/build-config.html#umdglobals.

---

update: 

1. Don't need to support `auto transform to camel case`, it was a limitation when implement with swc.
2. `non string type` might support it.

## Checklist

- [x] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/types/config/index.ts [hunk 1]
```diff
@@ -80,6 +80,7 @@ export interface LibConfig extends RsbuildConfig {
   footer?: BannerAndFooter;
   shims?: Shims;
   dts?: Dts;
+  umdName?: string;
```
Comment (Timeless0911) [inline packages/core/src/types/config/index.ts:83]: add to https://github.com/web-infra-dev/rslib/blob/main/packages/core/src/config.ts#L1100

File: tests/e2e/react-component/package.json [hunk 2]
```diff
@@ -2,8 +2,14 @@
   "name": "react-component-e2e",
   "version": "1.0.0",
   "private": true,
+  "scripts": {
+    "dev:bundle": "../../node_modules/.bin/rsbuild dev --environment=bundle",
+    "dev:bundle-false": "../../node_modules/.bin/rsbuild dev --environment=bundleFalse",
+    "dev:umd": "../../node_modules/.bin/rsbuild dev --environment=umd"
```
Comment (Timeless0911) [inline tests/e2e/react-component/package.json:8]: why not add rsbuild to devdeps in this package.json

Comment (fi3ework) [reply tests/e2e/react-component/package.json:8]: It's using @rsbuild/core in `../../` to start by the test runner.

File: packages/core/src/types/config/index.ts (hunk 1 above)
Comment (fi3ework) [reply packages/core/src/types/config/index.ts:83]: I changed the `omit` function, so it's required to add new keys here in the future.

File: tests/e2e/react-component/package.json (hunk 2 above)
Comment (Timeless0911) [reply tests/e2e/react-component/package.json:8]: using relative path to call bin file is a bit ugly

Comment (fi3ework) [reply tests/e2e/react-component/package.json:8]: Could refactor in other PR.

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**